pkg/controller/podautoscaler/metrics/metrics_client.go

```diff
@@ -119,6 +119,11 @@ func (h *HeapsterMetricsClient) GetCpuConsumptionAndRequestInMillis(namespace st
 	if err != nil {
 		return 0, 0, time.Time{}, fmt.Errorf("failed to get pod list: %v", err)
 	}
+
+	if len(podList.Items) == 0 {
+		return 0, 0, time.Time{}, fmt.Errorf("the pod list number is zero")
```

Comment:
The error message is confusing. Please change to: `empty pod list for selector: %v`